### PR TITLE
add buildsystem and sample code

### PR DESCRIPTION
**Pull request by zuntrax**
Some stuff @TheJJ did...

### CODE REVIEW

File: libnyan/nyan.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+/*
+TODO: static access
+ class LolObject : NyanObject {int health;};
+ NyanObject u = LolObject{1337};
+ int health = dynamic_cast<LolObject &>(u).health;
+
+dynamic access:
+ LolObject().get_attribute<int>("health");
+*/
```
Comment (TheJJ) [inline libnyan/nyan.h]: nope, we won't commit those comments and todo blocks.

**Comment by TheJJ** [comment]
obsolete.

**Comment by grindhold** [comment]
i recognize a pattern in sft software development :smiling_imp:

**Comment by zuntrax** [comment]
Well, @TheJJ dumped a bunch of files into a git repo owned by coop, without committing...
fak u jj

**Pull request closed without merging**